kythe/go/platform/tools/indexer_wrapper/main.go

```diff
@@ -0,0 +1,122 @@
+// Binary indexer_wrapper reads compilation units from the provided kzip(s) and
+// indexes them using language-specific indexers. Entry protos are written to
+// stdout.
+//
+// TODO(https://github.com/kythe/kythe/pull/3339): after the analyzer driver
+// interface is finalized, update this binary to use it rather than creating a
+// temporary kzip file for each indexer invocation.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"kythe.io/kythe/go/platform/analysis"
+	"kythe.io/kythe/go/platform/analysis/driver"
+	"kythe.io/kythe/go/platform/analysis/local"
+	"kythe.io/kythe/go/platform/delimited"
+
+	apb "kythe.io/kythe/proto/analysis_go_proto"
+)
+
+// delegatingAnalyzer provides an implementation of the CompilationAnalyzer
+// interface that forwards to a set of language-specific analyzers.
+type delegatingAnalyzer struct {
+	analyzers map[string]analysis.CompilationAnalyzer
+}
+
+// binPath returns the absolute path to an indexer given its name.
+func binPath(binname string) string {
+	return filepath.Join(*releaseDir, "indexers", binname)
+}
+
+func NewDelegatingAnalyzer(fetcher analysis.Fetcher) *delegatingAnalyzer {
+	// Command-line invocations for known/released indexer binaries.
+	cmds := map[string][]string{
+		"c++":       []string{binPath("cxx_indexer")},
+		"java":      []string{"java", "-jar", binPath("java_indexer.jar")},
+		"jvm":       []string{"java", "-jar", binPath("jvm_indexer.jar")},
+		"protobuf":  []string{binPath("proto_indexer"), "--index_file"},
+		"textproto": []string{binPath("textproto_indexer"), "--index_file"},
+		"go":        []string{binPath("go_indexer")},
+	}
+
+	analyzers := map[string]analysis.CompilationAnalyzer{}
+	for lang, cmd := range cmds {
+		analyzers[lang] = local.NewAnalyzer(cmd, fetcher)
+	}
+
+	return &delegatingAnalyzer{analyzers: analyzers}
+}
+
+func (a *delegatingAnalyzer) Analyze(ctx context.Context, req *apb.AnalysisRequest, f analysis.OutputFunc) error {
+	lang := req.Compilation.VName.Language
+	analyzer, ok := a.analyzers[lang]
+	if !ok {
+		return fmt.Errorf("unrecognized language: %q", lang)
+	}
+	log.Printf("analyzing compilation: {%+v}", req.Compilation.VName)
+
+	return analyzer.Analyze(ctx, req, f)
+}
+
+type driverContext struct{}
+
+func (driverContext) Setup(context.Context, driver.Compilation) error { return nil }
+func (driverContext) Teardown(ctx context.Context, comp driver.Compilation) error { return nil }
+
+// log any errors that happen during analysis.
+func (driverContext) AnalysisError(ctx context.Context, comp driver.Compilation, err error) error {
+	log.Printf("AnalysisError: %v", err)
+	return err
+}
+
+var (
+	releaseDir = flag.String("release_dir", "/opt/kythe", "Path to kythe release dir, which contains indexer binaries and more")
```

Comment:
I think the environment variable we use for this is `KYTHE_DIR`, it may be worthwhile to echo that as `--kythe_dir` here, though YMMV.